### ISSUE TITLE
add fs2

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#306]
this should have good cost/benefit ratio:
- nice clean modern build
- already published for 2.12.0-RC1
- no dependencies but ScalaCheck and ScalaTest
- lots of type fu in the code, to make scalac stammer and sweat

re fs2 vs scalaz-stream, see remarks at https://github.com/scala/community-builds/issues/246

**Comment by SethTisue** [comment]
fyi @pchiusano @mpilquist. I'll take a stab at this

**Comment by SethTisue** [comment]
```
[fs2] Processing subproject: docs
[fs2] [error] **** Missing dependency: the library org.tpolecat#tut-core is not provided (in space "default") by any project in this configuration file.
[fs2] [error] In order to control which version is used, please add the corresponding project to the build file
[fs2] [error] (or use "check-missing:false" to ignore (not recommended)).
[fs2] java.lang.RuntimeException: Required dependency not found
[fs2]   at scala.sys.package$.error(package.scala:27)
[fs2]   at com.typesafe.dbuild.plugin.DBuildRunner$$anonfun$fixModule$2.apply(DBuildRunner.scala:167)
[fs2]   at com.typesafe.dbuild.plugin.DBuildRunner$$anonfun$fixModule$2.apply(DBuildRunner.scala:142)
```

I'll just omit the `docs` project since we don't have Tut yet (fyi @tpolecat, I don't think anybody has actually ever tried to add it?)

**Comment by tpolecat** [comment]
The `tut-plugin` (which runs in sbt's 2.10 classloader) adds a dependency on `tut-core` (which is cross-built and runs in your program's classloader). It also adds a resolver so sbt can find `tut-core` on Bintray, which is where the plugin has to live. Someone else ran into this earlier today coincidentally. I guess I need to make `tut-core` find its way to Sonatype somehow? Where is `dbuild` looking?

**Comment by SethTisue** [comment]
dbuild wants to build every Scala dependency itself, from source, not pull binary artifacts from Bintray or anywhere else. so we'd need to add `tut-core` to the community build

**Comment by SethTisue** [comment]
note that sometimes adding a new project is trivial. example: https://github.com/scala/community-builds/pull/295/files — worked on the first try

**Comment by SethTisue** [comment]
fixed by #307

**Comment by mpilquist** [comment]
Awesome thanks Seth!
